File: CFD_Principles_and_Applications_Blazek/BookCode/C++/unstructured_2d/src/initConstants.cpp
/// @file initConstants.cpp
///
/// Initialization of constant values.
///
//*****************************************************************************
//
//  (c) J. Blazek, CFD Consulting & Analysis, www.cfd-ca.de
//  Created February 15, 2014
//  Last modification: July 2, 2014
// 
//=============================================================================
//
//  This program is free software; you can redistribute it and/or
//  modify it under the terms of the GNU General Public License
//  as published by the Free Software Foundation; either version 2
//  of the License, or (at your option) any later version.
//
//  This program is distributed in the hope that it will be useful,
//  but WITHOUT ANY WARRANTY; without even the implied warranty of
//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
//  GNU General Public License for more details.
//
//  You should have received a copy of the GNU General Public License
//  along with this program; if not, write to the Free Software
//  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
//
//*****************************************************************************

#include "defs.h"
#include "solver.h"

/// Initializes constants used by the solver.
///
void Solver::InitConstants()
{
  REAL gam1, rgas, temp, mach;

  gam1 = fluidProps.gamma - 1.0;
  rgas = gam1*fluidProps.cpgas/fluidProps.gamma;

  // external flow; it is assumed that "gamma" and "cpgas" specified in
  // the input file are valid for the complete far-field boundary
  if (bndConds.flowType == FlowType::External)
  {
    bndConds.alpha    = bndConds.alpha*RAD;
    bndConds.rhoinf   = bndConds.pinf/(rgas*bndConds.tinf);
    bndConds.qinf     = bndConds.machinf * SQRT(gam1*fluidProps.cpgas*bndConds.tinf);
    bndConds.uinf     = bndConds.qinf*COS(bndConds.alpha);
    bndConds.vinf     = bndConds.qinf*SIN(bndConds.alpha);
    precond.machRef2  = bndConds.machinf*bndConds.machinf;
    fluidProps.refRho = bndConds.rhoinf;
    fluidProps.refVel = bndConds.qinf;
    if (fluidProps.equsType == Equations::NavierStokes)
      fluidProps.refVisc = bndConds.rhoinf*bndConds.qinf*geometry.cref/fluidProps.renum;
    else
      fluidProps.refVisc = 0.0;
  }
  // internal flow
  else
  {
    temp = bndConds.ttinl * POW(bndConds.pout/bndConds.ptinl,gam1/fluidProps.gamma);
    mach = SQRT(2.0*((bndConds.ttinl/temp)-1.0)/gam1);

    bndConds.betainl = bndConds.betainl*RAD;
    bndConds.betaout = bndConds.betaout*RAD;
    precond.machRef2 = mach*mach;
    if (fluidProps.equsType == Equations::NavierStokes)
      fluidProps.refVisc = fluidProps.refRho*fluidProps.refVel*geometry.cref/fluidProps.renum;
    else
      fluidProps.refVisc = 0.0;
  }
}

File: CFD_Principles_and_Applications_Blazek/BookCode/Fortran/structured_2d/src/initConstants.f90
!> @file initConstants.f90
!!
!! Initialization of constant values.
!
! *****************************************************************************
!
!  (c) J. Blazek, CFD Consulting & Analysis, www.cfd-ca.de
!  Created February 25, 2014
!  Last modification: February 27, 2014
!
! *****************************************************************************
!
!  This program is free software; you can redistribute it and/or
!  modify it under the terms of the GNU General Public License
!  as published by the Free Software Foundation; either version 2
!  of the License, or (at your option) any later version.
!
!  This program is distributed in the hope that it will be useful,
!  but WITHOUT ANY WARRANTY; without even the implied warranty of
!  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
!  GNU General Public License for more details.
!
!  You should have received a copy of the GNU General Public License
!  along with this program; if not, write to the Free Software
!  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
!
! *****************************************************************************

!> Initializes constants used by the solver.
!!
subroutine InitConstants

  use ModDataTypes
  use ModGeometry
  use ModNumerics
  use ModPhysics
  implicit none

! local variables
  real(rtype) :: gam1, rgas

! *****************************************************************************

  pi  = 4.D0*Atan(1.D0)
  rad = 180.D0/pi

  if (kflow == 'E') then

! - external flow; it is assumed that "gamma" and "cpgas" specified in
!   the input file are valid for the complete far-field boundary

    gam1 = gamma - 1.D0
    rgas = gam1*cpgas/gamma

    alpha   = alpha/rad
    rhoinf  = pinf/(rgas*tinf)
    qinf    = machinf * Sqrt(gam1*cpgas*tinf)
    uinf    = qinf*Cos(alpha)
    vinf    = qinf*Sin(alpha)
    refrho  = rhoinf
    refvel  = qinf
    if (kequs == 'N') then
      refvisc = rhoinf*qinf*cref/renum
    else
      refvisc = 0.D0
    endif

  else

! - internal flow

    betainl = betainl/rad
    betaout = betaout/rad
    if (kequs == 'N') then
      refvisc = refrho*refvel*cref/renum
    else
      refvisc = 0.D0
    endif

  endif

end subroutine InitConstants
